pkg/gcrane/copy.go

```diff
@@ -63,8 +63,15 @@ func doCopy(args []string, recursive bool) {
 			log.Fatal(err)
 		}
 
-		if err := singleCopy(src, dst, srcAuth, dstAuth); err != nil {
-			log.Fatalf("failed to copy image: %v", err)
+		// First, try to copy as an index.
+		// If that fails, try to copy as an image.
+		// We have to try this second because fallback logic exists in the registry
+		// to convert an index to an  image.
+		// TODO(#388): Figure out which artifact is returned at runtime.
+		if err := copyIndex(src, dst, srcAuth, dstAuth); err != nil {
+			if err := copyImage(src, dst, srcAuth, dstAuth); err != nil {
```

Comment:
This version is a lot more readable, can we use this style in crane copy?

Reply:
Done